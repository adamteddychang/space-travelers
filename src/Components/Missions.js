import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchAllMissions } from '../Redux/missions/missions';
import Mission from './Mission';

function Missions() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);
  useEffect(() => {
    if (missions.length === 1) {
      dispatch(fetchAllMissions());
    }
  }, [dispatch]);
  return (
    <div className="missions">
      <table className="miss-table">
        <thead>
          <tr>
            <td className=" col-head">Mission</td>
            <td className=" col-head">Description</td>
            <td className=" col-head">Status</td>
            <td className=" col-head">Join</td>
          </tr>
        </thead>
        <tbody>
          {
          missions.map((mission) => (<Mission key={mission.mission_id} mission={mission} />))
        }
        </tbody>
      </table>
    </div>
  );
}

export default Missions;
