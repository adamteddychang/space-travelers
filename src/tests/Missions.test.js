import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Missions from '../Components/Missions';

describe('Mission Component test', () => {
  it('Renders Mission Row', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Missions />

      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
