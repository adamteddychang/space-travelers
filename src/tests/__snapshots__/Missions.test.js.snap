// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Component test Renders Mission Row 1`] = `
<div
  className="missions"
>
  <table>
    <thead>
      <tr>
        <td>
          Mission
        </td>
        <td>
          Description
        </td>
        <td>
          Status
        </td>
        <td>
          Are you Joined
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          <span
            className="available"
          >
            Available
          </span>
        </td>
        <td>
          <div>
            <button
              className="Join"
              onClick={[Function]}
              type="button"
            >
              Join
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
